cli-rs/src/main.rs

```diff
@@ -0,0 +1,351 @@
+        })
+        .init();
+
+    if args.markdown_help {
+        clap_markdown::print_help_markdown::<Cli>();
+        Ok(())
+    } else if let Some(command) = args.command {
+        run_command(command);
+        Ok(())
+    } else {
+        Cli::command().print_help()
+    }
+}
+
+fn split_inputs<T: FieldElement>(inputs: &str) -> Vec<T> {
+    inputs
+        .split(',')
+        .map(|x| x.trim())
+        .filter(|x| !x.is_empty())
+        .map(|x| x.parse::<BigUint>().unwrap().into())
+        .collect()
+}
+
+#[allow(clippy::print_stderr)]
+fn run_command(command: Commands) {
+    let result = match command {
+        Commands::Compile {
+            file,
+            field,
+            output_directory,
+            coprocessors,
+            continuations,
+        } => {
+            call_with_field!(compile_rust::<field>(
+                &file,
+                Path::new(&output_directory),
+                coprocessors,
+                continuations
+            ))
+        }
+        Commands::RiscvAsm {
+            files,
+            field,
+            output_directory,
+            coprocessors,
+            continuations,
+        } => {
+            assert!(!files.is_empty());
+            let name = if files.len() == 1 {
+                Cow::Owned(files[0].clone())
+            } else {
+                Cow::Borrowed("output")
+            };
+
+            call_with_field!(compile_riscv_asm::<field>(
+                &name,
+                files.into_iter(),
+                Path::new(&output_directory),
+                coprocessors,
+                continuations
+            ))
+        }
+        Commands::Execute {
+            file,
+            field,
+            inputs,
+            output_directory,
+            continuations,
+            witness,
+        } => {
+            call_with_field!(run_rust::<field>(
+                Path::new(&file),
+                split_inputs(&inputs),
+                Path::new(&output_directory),
+                continuations,
+                witness
+            ))
+        }
+    };
+    if let Err(errors) = result {
+        for error in errors {
+            eprintln!("{}", error);
+        }
+        std::process::exit(1);
+    }
+}
+
+#[allow(clippy::too_many_arguments)]
+fn compile_rust<F: FieldElement>(
+    file_name: &str,
+    output_dir: &Path,
+    coprocessors: Option<String>,
+    continuations: bool,
+) -> Result<(), Vec<String>> {
+    let runtime = match (coprocessors, continuations) {
+        (Some(list), _) => {
+            let runtime =
+                powdr_riscv::Runtime::try_from(list.split(',').collect::<Vec<_>>().as_ref())
+                    .unwrap();
+            if continuations {
+                runtime.with_poseidon()
+            } else {
+                runtime
+            }
+        }
+        (None, false) => powdr_riscv::Runtime::base(),
+        (None, true) => powdr_riscv::Runtime::base().with_poseidon(),
+    };
```

Comment:
```suggestion    let mut runtime = match coprocessors {
        Some(list) => {
            powdr_riscv::Runtime::try_from(list.split(',').collect::<Vec<_>>().as_ref()).unwrap()
        }
        None => powdr_riscv::Runtime::base(),
    };
    if continuations && !runtime.has_submachine("poseidon_gl") {
        runtime = runtime.with_poseidon();
    }
```

Cleaner IMO, and also fixed passing `-c --coprocessors poseidon_gl`, which previously complained that the poseidon coprocessor is added twice.